core/src/main/java/org/axonframework/config/EventHandlingConfiguration.java

```diff
@@ -451,34 +523,41 @@ public void initialize(Configuration config) {
             assignments.computeIfAbsent(processor, k -> new ArrayList<>()).add(handler);
         });
 
+        onInit.forEach(h -> h.accept(config));
         assignments.forEach((name, handlers) -> {
-            EventProcessor eventProcessor = eventProcessors.getOrDefault(name, defaultEventProcessorBuilder)
-                                                           .createEventProcessor(config, name, handlers);
-            interceptorsFor(config, name).forEach(eventProcessor::registerInterceptor);
-            initializedProcessors.add(eventProcessor);
+            config.eventProcessorRegistry().registerHandlerInvoker(name, c ->
+                    new SimpleEventHandlerInvoker(handlers,
+                                                  c.parameterResolverFactory(),
+                                                  getListenerInvocationErrorHandler(c, name)));
+            if (eventProcessorBuilders.containsKey(name)) {
+                eventProcessorBuilders.get(name).accept(config, handlers);
+            }
+            interceptorsFor(config, name).forEach(i -> config.eventProcessorRegistry()
+                                                             .registerHandlerInterceptor(name, c -> i));
         });
     }
 
     @Override
     public void start() {
-        initializedProcessors.forEach(EventProcessor::start);
+        // nothing to be started
     }
 
     @Override
     public void shutdown() {
-        initializedProcessors.forEach(EventProcessor::shutDown);
+        // nothing to be shut down
```

Comment:
Similar question as with the `start()` function.

Reply:
`EventProcessorRegistry` is responsible for shutting down the event processors.